service/scaladsl/server/src/main/scala/com/lightbend/lagom/scaladsl/server/LagomApplicationLoader.scala

```diff
@@ -248,11 +247,14 @@ private[server] object ActorSystemProvider {
 
   val logger = Logger(classOf[LagomApplication])
 
+  case object ApplicationShutdownReason extends CoordinatedShutdown.Reason
+
   /**
    * This is copied from Play's ActorSystemProvider, modified so we can inject json serializers
    */
   def start(config: Config, environment: Environment,
-            serializerRegistry: Option[JsonSerializerRegistry]): (ActorSystem, () => Future[Unit]) = {
+            serializerRegistry: Option[JsonSerializerRegistry]): (ActorSystem, () => Future[_]) = {
+
     val akkaConfig: Config = {
```

Comment:
Does this also need to include all of the code that was added in https://github.com/playframework/playframework/commit/f94cbaeba24864a0766a6983c9e2f6a0d5e8b6b9#diff-8e1dfac265f0cfecaa72a78c9a66f889R139 to handle `play.akka.shutdown-timeout`?